.github/workflows/nestbuildmatrix.yml

```diff
@@ -1,167 +1,368 @@
+      - name: "Restore apt cache"
+        uses: actions/cache@v2
+        env:
+          cache-name: "apt-cache"
+        with:
+          path: |
+            !/var/cache/apt/archives/lock
+            !/var/cache/apt/archives/partial
+            /var/cache/apt
+          key: ${{ runner.os }}-system-${{ env.cache-name }}-${{ hashFiles('**/environment.yml') }}
+          restore-keys: |
+            ${{ runner.os }}-system-${{ env.cache-name }}-
+            ${{ runner.os }}-system-
+
+      - name: "Install Linux system dependencies"
+        run: |
+          sudo apt-get update
+          #https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md
+          sudo apt-get install libltdl-dev libreadline6-dev libncurses5-dev libgsl0-dev python3-all-dev jq pycodestyle libpcre3 libpcre3-dev python2-dev libboost-all-dev
+          sudo apt-get install openmpi-bin libopenmpi-dev libgsl0-dev tcl8.6 tcl8.6-dev tk8.6-dev
+          sudo apt-get install libboost-filesystem-dev libboost-regex-dev libboost-wave-dev libboost-python-dev libboost-program-options-dev libboost-test-dev
+          sudo apt-get install vera++
+          sudo ldconfig
+          g++ --version
+
+      - name: "Restore pip cache"
+        env:
+          cache-name: "pip-cache"
+        uses: actions/cache@v2
+        with:
+          path: |
+            /opt/hostedtoolcache/Python/**/site-packages/*
+          key: ${{ runner.os }}-python-${{ env.cache-name }}-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-python-${{ env.cache-name }}-
+            ${{ runner.os }}-python-
+
+      - name: "Install Python dependencies"
+        run: |
+          python -m pip install --upgrade pip setuptools
+          python -c "import setuptools; print('package location:', setuptools.__file__)"
+          python -m pip install --force-reinstall --upgrade scipy 'junitparser>=2' numpy pytest pytest-timeout pytest-xdist mpi4py cython matplotlib terminaltables pandoc
+          python -c "import pytest; print('package location:', pytest.__file__)"
+          pip list
+
+      - name: "DEBUG: find commands"
+        run: |
+          echo "python3: $(which python3)"
+          echo "cython: $(which cython)"
+          echo "cythonize: $(which cythonize)"
+          echo "cython3: $(which cython3)"
+          echo "cython.py: $(which cython.py)"
+          echo "cythonize.py: $(which cythonize.py)"
+          find /opt/hostedtoolcache/Python/ -iname "*cython*"
+
+      - name: "Build NEST"
+        run: |
+          chmod +x build_support/ci_build.sh
+          ./build_support/ci_build.sh 2>&1 | tee ci_build.sh.log
+          python build_support/parse_build_log.py ci_build.sh.log ${{ github.workspace }}
+        env:
+          xNEST_BUILD_TYPE: ${{ matrix.xNEST_BUILD_TYPE }}
+          CHANGED_FILES: ${{ env.CHANGED_FILES }}
+          #get changed files: https://github.com/marketplace/actions/get-changed-files
```

Comment:
I left it in, because of the "changed files" magic that is happening. I'd rather remove the whole mechanism, but decided that it would be a bit much for this PR. (A follow up PR is already in preparation)